app/models/label/LabelTable.scala

```diff
@@ -269,12 +273,73 @@ object LabelTable {
+        |	FROM sidewalk.label as lb1, sidewalk.audit_task as at,
+        |       sidewalk.user as u, sidewalk.label_point as lp,
+        |				(SELECT lb.label_id, lb.gsv_panorama_id, lbt.label_type, lbt.description as label_type_desc, sev.severity,
+        |               COALESCE(prob_temp.temporary_problem,'FALSE') as temp_problem,
+        |               prob_desc.description
+        |					FROM label as lb
+        |				LEFT JOIN sidewalk.label_type as lbt
+        |					ON lb.label_type_id = lbt.label_type_id
+        |				LEFT JOIN sidewalk.problem_severity as sev
+        |					ON lb.label_id = sev.label_id
+        |				LEFT JOIN sidewalk.problem_description as prob_desc
+        |					ON lb.label_id = prob_desc.label_id
+        |				LEFT JOIN sidewalk.problem_temporariness as prob_temp
+        |					ON lb.label_id = prob_temp.label_id
+        |				) AS lb_big
+        |WHERE u.user_id = ? and
+        |      lb1.deleted = FALSE and lb1.audit_task_id = at.audit_task_id and
+        |      lb1.label_id = lb_big.label_id and at.user_id = u.user_id and lb1.label_id = lp.label_id
+        |	ORDER BY lb1.label_id DESC
+        | LIMIT ?""".stripMargin
+    )
+    selectQuery((userId, takeN)).list.map(label => LabelMetadata.tupled(label))
+  }
+
+  def retrieveSingleLabelMetadata(labelId: Int): LabelMetadata = db.withSession { implicit session =>
+    val selectQuery = Q.query[Int,(Int, String, Float, Float, Int, Int, Int, Int, Int,
+      Int, String, String, Option[java.sql.Timestamp], String, String, Option[Int], Boolean,
+      Option[String])](
+      """SELECT lb1.label_id, lb1.gsv_panorama_id, lp.heading, lp.pitch, lp.zoom, lp.canvas_x, lp.canvas_y,
+        |       lp.canvas_width, lp.canvas_height, lb1.audit_task_id, u.user_id, u.username, lb1.time_created,
+        |       lb_big.label_type, lb_big.label_type_desc, lb_big.severity, lb_big.temp_problem, lb_big.description
+        |	FROM sidewalk.label as lb1, sidewalk.audit_task as at,
+        |       sidewalk.user as u, sidewalk.label_point as lp,
+        |				(SELECT lb.label_id, lb.gsv_panorama_id, lbt.label_type, lbt.description as label_type_desc, sev.severity,
+        |               COALESCE(prob_temp.temporary_problem,'FALSE') as temp_problem,
+        |               prob_desc.description
+        |					FROM label as lb
+        |				LEFT JOIN sidewalk.label_type as lbt
+        |					ON lb.label_type_id = lbt.label_type_id
+        |				LEFT JOIN sidewalk.problem_severity as sev
+        |					ON lb.label_id = sev.label_id
+        |				LEFT JOIN sidewalk.problem_description as prob_desc
+        |					ON lb.label_id = prob_desc.label_id
+        |				LEFT JOIN sidewalk.problem_temporariness as prob_temp
+        |					ON lb.label_id = prob_temp.label_id
+        |				) AS lb_big
+        |WHERE lb1.label_id = ? and
+        |      lb1.deleted = FALSE and lb1.audit_task_id = at.audit_task_id and
```

Comment:
deleted shouldn't be included in the query since we want metadata for label irrespective of whether the label is deleted or not